Add an option of restricted output characters for hjr0001 (#1138)
opencv/open_model_zoo: demos/python_demos/handwritten_japanese_recognition_demo/README.md

                        Optional. Number of inference iterations
  -cl CHARLIST, --charlist CHARLIST
                        Path to the decoding char list file
  -dc DESIGNATED_CHARACTER, --designated_character DESIGNATED_CHARACTER
                        Path to the designated character file
  -tk TOP_K, --top_k TOP_K
                        Top k steps in looking up the decoded character, until
                        a designated one is found
```

For example:
```
python handwritten_japanese_recognition_demo.py -i data/test.png -m path/to/ir_xml/model.xml

```

> When the designated_character argument is provided, the script will check Top k steps in looking up the decoded character, until a designated one is found. By doing so, the output character will be restricted to a designated region. K is set to 20 by default.

Reviewer: ```suggestion
When the `designated_character` argument is provided, the script will check Top k steps in looking up the decoded character, until a designated one is found. By doing so, the output character will be restricted to a designated region. K is set to 20 by default.
```

> For example, if we want to restrict the output character to a digit and hyphen circumstance, we need to provide the path to the designated character file, then the command line will be like:

Reviewer 2: ```suggestion
For example, if we want to restrict the output character only to digits and hyphen circumstance, we need to provide the path to the designated character file and set K to the number of rows in charlist file, then the command line will be like:
```
The initial version misled me. I expected that only the chars from `digit_hyphen.txt` will appear, but other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements.

Author: Actually, your initial expectation is right. This pull request is based on the charlist file.  But if an additional designated_characters file is provided, this script will perform a **post**-filtering processing, which means it is possible that other characters are still allowed if none of digit_hyphen.txt is in first K chosen elements. Thus, the -tk argument has little to do with the charlist file, instead it is an extra argument for post-processing.

Reviewer 2: I think I was unclear with my comment. I meant that

I expected that only the chars from `digit_hyphen.txt` will appear, but *the reality turned out to be that* other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements.

> post-filtering processing, which means it is possible that other characters are still allowed if none of digit_hyphen.txt is in first K chosen elements.

So that's what I was trying to point out, but your version is clearer :) Could you add it to readme?

Author: Sure. Thanks your careful review

```

> python handwritten_japanese_recognition_demo.py -i data/test.png -m path/to/ir_xml/model.xml -dc data/digit_hyphen.txt

Reviewer 2: ```suggestion
python handwritten_japanese_recognition_demo.py -i data/test.png -m path/to/ir_xml/model.xml -dc data/digit_hyphen.txt -tk 4442
```

Author: Such an large `-tk 4442` does not make sense here, as it is a parameter for post-processing.

                        Optional. Number of inference iterations
  -cl CHARLIST, --charlist CHARLIST
                        Path to the decoding char list file
  -dc DESIGNATED_CHARACTERS, --designated_characters DESIGNATED_CHARACTERS
                        Optional. Path to the designated character file
  -tk TOP_K, --top_k TOP_K
                        Optional. Top k steps in looking up the decoded
                        character, until a designated one is found
```

For example:
```
python handwritten_japanese_recognition_demo.py -i data/test.png -m path/to/ir_xml/model.xml

```
When the `designated_characters` argument is provided, if the output character is not included in the designated characters, the script will check Top k steps in looking up the decoded character, until a designated one is found. By doing so, the output character will be restricted to a designated region. K is set to 20 by default.

> For example, if we want to restrict the output character to an only digits and hyphens circumstance, we need to provide the path to the designated character file, e.g. `digit_hyphen.txt`. Then the script will perform a post-filtering processing on the ouptut characters, but please note that it is possible that other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements.

Reviewer 3: ```suggestion
For example, if we want to restrict the output characters to only digits and hyphens, we need to provide the path to the designated character file, e.g. `digit_hyphen.txt`. Then the script will perform a post-filtering processing on the ouptut characters, but please note that it is possible that other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements.
```
